src/Compilers/CSharp/Portable/Symbols/MethodSymbolExtensions.cs

```diff
@@ -300,5 +301,25 @@ public static bool IsGenericTaskReturningAsync(this MethodSymbol method, CSharpC
             return method.IsAsync
                 && method.ReturnType.IsGenericTaskType(compilation);
         }
+
+        internal static CSharpSyntaxNode ExtractReturnTypeSyntax(this MethodSymbol method)
+        {
+            method = method?.PartialImplementationPart ?? method;
```

Comment:
>method = method?.PartialImplementationPart ?? method; [](start = 12, length = 53)

Do we ever call this method with null? Also, we probably should prefer the definition part for a partial method.